core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -455,8 +455,29 @@ private void doStop(CompletableFuture<Void> future,
         }
 
         private void finishDoStop(CompletableFuture<Void> future) {
-            // TODO(trustin): Add shutdownBlockingTaskExecutorOnStop
-            // TODO(trustin): Count the pending blocking tasks and wait until it goes zero.
+            if (config.shutdownBlockingTaskExecutorOnStop()) {
+                final ExecutorService executor;
+                if (config.blockingTaskExecutor() instanceof InterminableExecutorService) {
```

Comment:
nit: You could extract `config.blockingTaskExecutor()` as a variable and use it below.

Reply:
Indeed. @geminiKim, would you mind cleaning this up a little bit?